### PR TITLE
Moving Projectlockfile Apis from Microsoft.DotNet.ProjectModel to NuGet.ProjectModel

### PR DESCRIPTION
**Pull request by livarcocc**
@eerhardt @brthor @Sridhar-MS @emgarten

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectJsonMigration/ProjectDependencyFinder.cs [hunk 1]
```diff
@@ -60,6 +60,7 @@ public IEnumerable<ProjectDependency> ResolveProjectDependencies(string projectD
 
                 if (!possibleProjectDependencies.TryGetValue(projectExportName, out projectDependency))
                 {
+                    Console.WriteLine($"Don't have it: {projectExportName} {projectExport.Library.Identity.Type}");
```
Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/ProjectDependencyFinder.cs]: Migration trace.

Comment (livarcocc) [inline src/Microsoft.DotNet.ProjectJsonMigration/ProjectDependencyFinder.cs]: Nvm, this was for debugging. Will remove.

File: src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs [hunk 2]
```diff
@@ -412,11 +415,12 @@ private IEnumerable<AnalyzerReference> GetAnalyzerReferences(PackageDescription
             var analyzers = package
                 .PackageLibrary
                 .Files
-                .Where(path => path.StartsWith("analyzers" + Path.DirectorySeparatorChar) &&
+                .Where(path => path.StartsWith("analyzers" + LockFile.DirectorySeparatorChar) &&
```
Comment (brthor) [inline src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs]: Try changing this back to the current platform and see if your analyzer bug goes away.

Comment (brthor) [inline src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs]: Or maybe make it look for any directory separator char.

File: src/dotnet/commands/dotnet-build/CompilerIOManager.cs [hunk 3]
```diff
@@ -109,12 +109,7 @@ private static void AddLockFile(ProjectContext project, List<string> inputs)
                 throw new InvalidOperationException(errorMessage);
             }
 
-            inputs.Add(project.LockFile.LockFilePath);
-
-            if (project.LockFile.ExportFile != null)
```
Comment (eerhardt) [inline src/dotnet/commands/dotnet-build/CompilerIOManager.cs]: Isn't this still necessary?

**Comment by eerhardt** [comment]
:shipit:

File: src/dotnet/commands/dotnet-build/CompilerIOManager.cs (hunk 3 above)
Comment (livarcocc) [inline src/dotnet/commands/dotnet-build/CompilerIOManager.cs]: This is for the msbuild fragment, right? Which we don't need anymore.

**Comment by livarcocc** [comment]
@dotnet-bot Test this please

File: src/dotnet/commands/dotnet-build/CompilerIOManager.cs (hunk 3 above)
Comment (eerhardt) [inline src/dotnet/commands/dotnet-build/CompilerIOManager.cs]: > Which we don't need anymore.

We need it for .xproj => .csproj references right? Aren't we still supporting project.json/.xproj in the current code?  If not, we should remove the project-model-server again.

**Pull request merged**